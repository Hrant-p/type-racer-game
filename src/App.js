import React from 'react';
import './App.scss';
import { BrowserRouter, Switch, Route } from 'react-router-dom';
import PrivateRoute from './container/PrivateRoute/PrivateRoute';
import Home from './container/Home/Home';
import Login from './container/Login/Login';
import SignUp from './container/SignUp/SignUp';
import Nav from './components/Nav/Nav';
import Footer from './components/Footer/Footer';
import About from './components/About/About';
import NotFound from './components/NotFound/NotFound';

const App = () => {
  return (
    <BrowserRouter>
      <Nav />
      <Switch>
        <PrivateRoute path="/home" component={Home} exact />
        <Route path="/(login||)" component={Login} exact />
        <Route path="/registration" component={SignUp} exact />
        <Route path="/about" component={About} exact />
        <Route component={NotFound} />
      </Switch>
      <Footer />
    </BrowserRouter>
  );
};

export default App;
