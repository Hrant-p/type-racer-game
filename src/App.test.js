import React from 'react';
import renderer from 'react-test-renderer';
import App from './App';

it('App correctly renders after relaunch', () => {
  const tree = renderer
    .create(<App
      onClick={jest.fn()}
    />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
