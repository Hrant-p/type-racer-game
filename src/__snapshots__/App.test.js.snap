// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App correctly renders after relaunch 1`] = `
Array [
  <nav
    className="head-block"
  >
    <ul>
      <li>
        <a
          href="/"
          onClick={[Function]}
        >
          <img
            alt="icon"
            src="icon.png"
          />
        </a>
      </li>
      <li>
        <a
          href="/home"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li>
        <a
          href="/login"
          onClick={[Function]}
        >
          Login
        </a>
      </li>
      <li>
        <a
          href="/registration"
          onClick={[Function]}
        >
          SignUp
        </a>
      </li>
      <li>
        <a
          href="/about"
          onClick={[Function]}
        >
          About
        </a>
      </li>
    </ul>
  </nav>,
  <div
    className="loader"
  />,
  <div
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "indigo",
        "display": "flex",
        "justifyContent": "center",
      }
    }
  >
    <p
      className="sc-AykKC jCAioK"
    >
      2020
       
      Â© âœ¨ Created with React & Redux By Hrant Poghosyan ðŸ‘Œ
    </p>
  </div>,
]
`;
